sdk/tests/wallet/balance.rs

```diff
@@ -165,6 +165,76 @@ async fn balance_expiration() -> Result<()> {
     tear_down(storage_path)
 }
 
+#[ignore]
+#[tokio::test]
+async fn addresses_balance() -> Result<()> {
+    let storage_path = "test-storage/addresses_balance";
+    setup(storage_path)?;
+
+    let wallet = make_wallet(storage_path, None, None).await?;
+
+    let account_0 = &create_accounts_with_funds(&wallet, 1).await?[0];
+    let account_1 = wallet.create_account().finish().await?;
+    let addresses_0 = account_0.addresses_with_unspent_outputs().await?;
+    let acc_1_addr = &account_1.generate_ed25519_addresses(1, None).await?[0];
+
+    let balance_0: Balance = account_0
+        .addresses_balance(addresses_0.iter().map(|a| a.address()).collect())
+        .await?;
+    let balance_0_sync = account_0.balance().await?;
+    let to_send = balance_0.base_coin().available();
+
+    // Check if 0 has balance and sync() and address_balance() match
+    assert!(to_send > 0);
+    assert_eq!(balance_0, balance_0_sync);
+
+    // Make sure 1 is empty
+    let balance_1: Balance = account_1.sync(None).await?;
+    assert_eq!(balance_1.base_coin().available(), 0);
+
+    // Send to 1
+    let tx = account_0.send(to_send, acc_1_addr.address(), None).await?;
+    // Balance should update without sync
+    let balance_0: Balance = account_0
+        .addresses_balance(addresses_0.iter().map(|a| a.address()).collect())
+        .await?;
+    let balance_0_sync = account_0.balance().await?;
+    assert_eq!(balance_0.base_coin().available(), 0);
+    assert_eq!(balance_0, balance_0_sync);
+
+    account_0
+        .retry_transaction_until_included(&tx.transaction_id, None, None)
+        .await?;
+    account_1.sync(None).await?;
+
+    // Balance should have transferred entirely
+    let balance_1: Balance = account_1.addresses_balance(vec![acc_1_addr.address()]).await?;
+    let balance_1_sync = account_1.balance().await?;
+    assert!(balance_1.base_coin().available() > 0);
+    assert_eq!(balance_1, balance_1_sync);
+
+    // Internal transfer on account 1
+    let acc_1_addr_2 = &account_1.generate_ed25519_addresses(1, None).await?[0];
+
+    let tx = account_1.send(to_send / 2, acc_1_addr_2.address(), None).await?;
+    account_1
+        .retry_transaction_until_included(&tx.transaction_id, None, None)
+        .await?;
+    let balance_1_sync = account_1.sync(None).await?;
+
+    // Check the new address
+    let balance_1: Balance = account_1.addresses_balance(vec![acc_1_addr_2.address()]).await?;
+    assert_eq!(to_send / 2, balance_1.base_coin().available());
+
+    // Check old and new together
+    let balance_1_total: Balance = account_1
```

Comment:
Just curious, why explicit the type?

Reply:
argh, my ide keeps adding that when i ctrl+click on the type preview.